### ISSUE TITLE
Array is treated as object in certain setup

### ISSUE THREAD

**Issue by variant77** [rtfeldman/seamless-immutable#146]
It happens only on one of my computers and I could not understand why but Immutable fails to detect arrays as such and treats them like regular objects.

The simple reconstruction:
```
Immutable = require('seamless-immutable')
Immutable([1,2,3])
//output is:
{ '0': 1, '1': 2, '2': 3 }
```

**Comment by tommoor** [comment]
This also happens in our environment - really frustrating as it makes it impossible to create Immutable arrays. I'd love to see your PR specs fixed up and merged!

**Comment by variant77** [comment]
Thanks for letting more know I was not the only one affected. At some point
I was losing my mind over it.

My PR seem to be just fine, all the tests poss when I run them locally.
There is some configuration error in the repo that causes all PRs to fail.

If you want to use the fixed version before the PR is merged you can use my
fork for now and subscribe to this thread to know when to switch back.

**Comment by rtfeldman** [comment]
@tommoor FYI @variant77's fix has been merged and released as 6.1.2!
